shared/protocol/src/main/java/io/pravega/shared/segment/StreamSegmentNameUtils.java

```diff
@@ -366,4 +375,12 @@ public static boolean isTableSegment(String qualifiedName) {
         return tokens[1].equals(TABLES);
     }
     // endregion
+    
+    // region watermark
+    public static String getMarkSegmentForStream(String stream) {
```

Comment:
We should probably call this `getMarkStreamForStream` because there still will be a segment number associated with the actual segment.